Signal/src/contact/OWSContactsManager.m

```diff
@@ -399,12 +399,26 @@ - (BOOL)phoneNumber:(PhoneNumber *)phoneNumber1 matchesNumber:(PhoneNumber *)pho
 
 #pragma mark - Whisper User Management
 
+- (BOOL)hasNameInSystemContactsForRecipientId:(NSString *)recipientId
+{
+    return [self cachedDisplayNameForRecipientId:recipientId] != nil;
+}
+
 - (NSString *)unknownContactName
 {
     return NSLocalizedString(@"UNKNOWN_CONTACT_NAME",
                              @"Displayed if for some reason we can't determine a contacts phone number *or* name");
 }
 
+- (nullable NSString *)formattedProfileNameForRecipientId:(NSString *)recipientId
+{
+    NSString *_Nullable profileName = [self.profileManager profileNameForRecipientId:recipientId];
+    if (profileName == nil) {
+        return nil;
+    }
+    return [@"~" stringByAppendingString:profileName];
```

Comment:
Hm. Any insight into what this should be? 

We could localize the format, but it will be a bit unwieldy for the translators. Do you think this is reasonable?

/* Format string for displaying a profile name below a message bubble.  "%1$@ embeds the contact's {{phone number}} and  %2$@ embeds their {{profile name}} */
"PROFILE_NAME_FORMAT" = @"%1$@ ~%2$@"

Or should we just flip it when RTL?